server/apikeys.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ApiKey struct {
+	ID        uint      `gorm:"primarykey" json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+	Value     string    `gorm:"not null" json:"value"`
+	UserID    uint      `gorm:"not null" json:"-"`
+	Note      string    `gorm:"not null" json:"note"`
+}
+
+type ApiKeyRequest struct {
+	Note string `json:"note,required"`
```

Comment:
I'm getting this warning: `unknown JSON option "required" (SA5008)` from go-staticcheck